File: Java/src/com/dbc/algorithm/Array/PascalsTriangleII.java
package com.dbc.algorithm.Array;

import java.util.ArrayList;
import java.util.List;

public class PascalsTriangleII {
    public static List<Integer> getRow(int rowIndex) {
        List<Integer> result = new ArrayList<>();
        result.add(1);
        int remain = 0;
        for (int i = 0; i < rowIndex; i++) {
            for (int j = 1; j < i + 1; j++) {
                if (j == 1) {
                    remain = result.get(j);
                    result.set(j, result.get(j) + result.get(j - 1));
                } else {
                    int temp = result.get(j);
                    result.set(j, result.get(j) + remain);
                    remain = temp;
                }
            }
            result.add(1);
        }
        return result;
    }
}

File: src/tmp/Python/array/pascals_triangle_ii.py
from typing import List

def getRow(rowIndex: int) -> List[int]:
    result = [1]
    for i in range(rowIndex):
        for j in range(1, i + 1):
            
            if j == 1:
                temp = result[j]
                result[j] = result[j - 1] + result[j]
            else:
                remain = result[j]
                result[j] = result[j] + temp
                temp = remain
            
        result.append(1)
    
    return result

if __name__ == "__main__":
    print(getRow(1))

